lib/rails_multitenant/global_context_registry.rb

```diff
@@ -91,6 +91,10 @@ def as_current_id(id)
           self.current_id = old_id
         end
 
+        def as_current(model, &block)
+          as_current_id(model&.id, &block)
```

Comment:
This will end up looking the model again if `model_class.current` is referenced. Can we avoid this?

Reply:
Yes, I will take a look and see how much work that would be.